checks/raw/branch_protection.go

```diff
@@ -57,78 +76,47 @@ func BranchProtection(c clients.RepoClient) (checker.BranchProtectionsData, erro
 			continue
 		}
 
-		// Try to resolve the branch name.
-		b, err := branchesMap.getBranchByName(release.TargetCommitish)
-		if err != nil {
-			// If the commitish branch is still not found, fail.
-			return checker.BranchProtectionsData{}, err
+		if branches.contains(release.TargetCommitish) ||
+			branches.contains(branchRedirect(release.TargetCommitish)) {
+			continue
 		}
 
-		// Branch is valid, add to list of branches to check.
-		checkBranches[*b.Name] = true
-	}
-
-	// Add default branch.
-	defaultBranch, err := c.GetDefaultBranch()
-	if err != nil {
-		return checker.BranchProtectionsData{}, fmt.Errorf("%w", err)
-	}
-	defaultBranchName := getBranchName(defaultBranch)
-	if defaultBranchName != "" {
-		checkBranches[defaultBranchName] = true
-	}
-
-	rawData := checker.BranchProtectionsData{}
-	// Check protections on all the branches.
-	for b := range checkBranches {
-		branch, err := branchesMap.getBranchByName(b)
+		// Get the associated release branch.
+		branchRef, err := c.GetBranch(release.TargetCommitish)
 		if err != nil {
-			if errors.Is(err, errInternalBranchNotFound) {
-				continue
-			}
-			return checker.BranchProtectionsData{}, err
+			return checker.BranchProtectionsData{},
+				fmt.Errorf("error during GetBranch(%s): %w", release.TargetCommitish, err)
+		}
+		if branches.add(branchRef) {
+			continue
 		}
 
-		rawData.Branches = append(rawData.Branches, *branch)
+		// Couldn't find the branch check for redirects.
```

Comment:
what's a redirect?

Reply:
See https://github.com/google/go-github/issues/1895. Basically, if `master` has been renamed to `main`, some API responses will still return the older branch name.